### PR TITLE
Ask va api create submitter profile class

### PR DESCRIPTION
**Pull request by Khoa-V-Nguyen**
## Summary:

This PR introduces key improvements to the SubmitterProfile class and implements a new locations.yml file to store states and countries for better modularity and maintainability.

- SubmitterProfile Refactor: The SubmitterProfile class has been refactored to improve modularity and clarity by breaking down the profile building logic into smaller, focused methods (base_profile, contact_info, school_info, and service_info). Helper methods such as country_data and state_data were added to manage location-related information more effectively. Additionally, the Translator class is used to handle field translation, improving readability and future extensibility.
- Locations YAML File: A new locations.yml file has been created to store state and country information, centralizing the location data management and making it easier to maintain and update.

This refactor enhances the overall structure and maintainability of the profile-building process while improving the flexibility for future development.

## Related issue(s)

- https://github.com/department-of-veterans-affairs/ask-va/issues/1374

## Testing done

- [x] *New code is covered by unit tests*

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [x]  Events are being sent to the appropriate logging solution
- [x]  Documentation has been updated (link to documentation)
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x]  Feature/bug has a monitor built into Datadog (if applicable)

### CODE REVIEW

**Review by hemeshvpatel — COMMENTED** [review]
Lgtm

**Review by hemeshvpatel — APPROVED** [review]
Lgtm

File: modules/ask_va_api/app/lib/ask_va_api/optionset/retriever.rb [hunk 1]
```diff
@@ -7,26 +7,8 @@ class OptionsetRetrieverError < StandardError; end
     class Retriever < BaseRetriever
       attr_reader :name
```
Comment (stevenjcumming) [inline modules/ask_va_api/app/lib/ask_va_api/optionset/retriever.rb:8]: Can `name` be removed too? It looks like it's also referenced in modules/ask_va_api/spec/app/lib/ask_va_api/optionset/retriever_spec.rb on line 11

Comment (Khoa-V-Nguyen) [reply modules/ask_va_api/app/lib/ask_va_api/optionset/retriever.rb]: Good catch. Updated

**Review by stevenjcumming — APPROVED** [review]

**Pull request merged**